File: LPAv/Trabalho 9 - Python/questao5.py
import re

stringDeEntrada = input()
while (stringDeEntrada != "####"):
    if (re.match("^http[s]{0,1}:\\/\\/[a-zA-Z-_]+(\\.[a-zA-Z-_]+)+(\\/[a-zA-Z-_]+)*(\\.[a-zA-Z-_]+){0,1}$", stringDeEntrada)):
        print("SIM")
    else:
        print("NAO")
    stringDeEntrada = input()

File: LPAv/Trabalho 9 - C++/questao5.cpp
#include <iostream>
#include <stdio.h>
#include <string>
#include <regex>
using namespace std;

int main()
{
    string expressaoRegular;
    cin >> expressaoRegular;
    while (expressaoRegular != "####")
    {
        if (regex_match(expressaoRegular, regex("^http[s]{0,1}:\\/\\/[a-zA-Z-_]+(\\.[a-zA-Z-_]+)+(\\/[a-zA-Z-_]+)*(\\.[a-zA-Z-_]+){0,1}$")))
        {
            printf("SIM\n");
        }
        else
        {
            printf("NAO\n");
        }
        cin >> expressaoRegular;
    }
}